code/game/gamemodes/miniantags/pulsedemon/pulsedemon_interactions.dm

```diff
@@ -0,0 +1,272 @@
+
+/mob/living/simple_animal/pulse_demon/proc/check_valid_cell_charger(obj/machinery/cell_charger/C)
+	return C.charging
+
+/mob/living/simple_animal/pulse_demon/proc/finish_hijack_cell_charger(obj/machinery/cell_charger/C)
+	to_chat(src, "<span class='notice'>You are now inside [C.charging]. Click on a hijacked APC to return.</span>")
+	forceMove(C.charging)
+	C.charging.rigged = TRUE
+
+/obj/machinery/recharge_station/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	user.forceMove(src)
+	if(user.check_valid_recharge_station(src))
+		var/mob/living/silicon/robot/R = occupant
+		if(R in user.hijacked_robots)
+			user.finish_hijack_recharge_station(src)
+			return
+		if(!user.pb_helper.start(user, src, user.hijack_time, TRUE, PD_HIJACK_CB(user, check_valid_recharge_station), PD_HIJACK_CB(user, finish_hijack_recharge_station), PD_HIJACK_FAIL(user)))
+			to_chat(user, "<span class='warning'>You are already performing an action!</span>")
+		else
+			user.do_hijack_notice(occupant)
+			to_chat(R, "<span class='danger'>ALERT: ELECTRICAL MALEVOLENCE DETECTED, TARGETING SYSTEMS HIJACK IN PROGRESS</span>")
+	else
+		to_chat(user, "<span class='warning'>There is no silicon-based occupant inside. Click again to retry.</span>")
+
+/mob/living/simple_animal/pulse_demon/proc/check_valid_recharge_station(obj/machinery/recharge_station/R)
+	return R.occupant && istype(R.occupant, /mob/living/silicon/robot)
+
+/mob/living/simple_animal/pulse_demon/proc/finish_hijack_recharge_station(obj/machinery/recharge_station/S)
+	do_hijack_robot(S.occupant)
+
+/mob/living/simple_animal/pulse_demon/proc/do_hijack_robot(mob/living/silicon/robot/R)
+	to_chat(src, "<span class='notice'>You are now inside [R]. Click on a hijacked APC to return.</span>")
+	forceMove(R)
+	current_robot = R
+	if(!(R in hijacked_robots))
+		hijacked_robots += R
+		to_chat(R, "<span class='userdanger'>TARGETING SYSTEMS HIJACKED, REPORT ALL UNWANTED ACTIVITY</span>")
+
+/obj/machinery/camera/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	if(user.loc != src)
+		user.forceMove(src)
+		to_chat(user, "<span class='notice'>You jump towards [src]. Click on a hijacked APC to return.</span>")
+
+// see pulse_demon/say
+/obj/machinery/hologram/holopad/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	if(user.loc != src)
+		user.forceMove(src)
+		to_chat(user, "<span class='notice'>You jump towards [src]. You can now communicate via the holopad's speaker. Click on a hijacked APC to return.</span>")
+
+/obj/item/radio/attack_pulsedemon(mob/living/simple_animal/pulse_demon/user)
+	if(user.loc != src)
+		user.forceMove(src)
+		to_chat(user, "<span class='notice'>You jump towards [src]. You can now communicate via radio. Click on a hijacked APC to return.</span>")
+	else
+		attack_ai(user)
+
+// for overrides on bots
+/atom/proc/attack_integrated_pulsedemon(mob/living/simple_animal/pulse_demon/user, atom/A)
+	return
```

Comment:
unless this is being used for anything other than bots, don't define this on the atom level (at a minimum it needs to be atom/moveable :/). You should just scope this to `/mob/living/simple_animal/bot` and typecheck where need be, it already looks like you do anyway